### DISCUSSION TITLE
How to use DigestAlgorithm in Sign method?

### DISCUSSION THREAD

**Discussion by saigkill** [cake-build/cake#3810] [Q&A] [answered]
Hello team,

i used that config:

Task("Sign20348")
	.IsDependentOn("Publish20348")
	.Does(() =>
	{
		var file = new FilePath("./bin/publish-20348/WayFair-Label-App.exe");
		Sign(file, new SignToolSignSettings {
		TimeStampUri = new Uri("http://timestamp.digicert.com"),
		CertPath = "MarcosSoftwareKey.pfx",
		Password = "pioneers"
		});		
	});

I got the issuse: SignTool Error: No file digest algorithm specified. Please specify the digest algorithm with the /fd flag. Using /fd SHA256 is recommended and more secure than SHA1. Calling signtool with /fd sha1 is equivalent to the previous behavior. In order to select the hash algorithm used in the signing certificate's signature, use the /fd certHash option.

I found in the doku: https://cakebuild.net/api/Cake.Common.Tools.SignTool/SignToolDigestAlgorithm/

So i tried:

Task("Sign20348")
	.IsDependentOn("Publish20348")
	.Does(() =>
	{
		var file = new FilePath("./bin/publish-20348/WayFair-Label-App.exe");
		Sign(file, new SignToolSignSettings {
		TimeStampUri = new Uri("http://timestamp.digicert.com"),
		CertPath = "MarcosSoftwareKey.pfx",
		Password = "pioneers"
		DigestAlgorithm = "1"
		});		
	});

But i got: C:/Users/Sascha/OneDrive/source/repos/marcos software/WayFairLabelApp/WayFair-Label-App/build.cake(121,21): error CS0029: Der Typ "string" kann nicht implizit in "Cake.Common.Tools.SignTool.SignToolDigestAlgorithm" konvertiert werden.

I also tried 1 instead of "1". But that doesnt work.

Maybe its too late for me to understand it right. Maybe anyone can help out?

Greetings Sascha

**Comment by nils-a** [comment, marked as answer]
@saigkill The value of the `DigestAlgorithm` property is of type [`SignToolDigestAlgorithm`](https://cakebuild.net/api/Cake.Common.Tools.SignTool/SignToolDigestAlgorithm/), so possible values are `SignToolDigestAlgorithm.Sha1` or `SignToolDigestAlgorithm.Sha256`.

Try:

```cs
Task("Sign20348")
 .IsDependentOn("Publish20348")
 .Does(() =>
{
  var file = new FilePath("./bin/publish-20348/WayFair-Label-App.exe");
  Sign(file, new SignToolSignSettings 
  {
    TimeStampUri = new Uri("http://timestamp.digicert.com"),
    CertPath = "MarcosSoftwareKey.pfx",
    Password = "pioneers",
    DigestAlgorithm = SignToolDigestAlgorithm.Sha256
  });
});
```

**Reply by saigkill** [reply]
Indeed, it was too late. Thanks for anwering. It works now as expected :-)
